mesheryctl/internal/cli/root/app/import.go

```diff
@@ -0,0 +1,297 @@
+		if err != nil {
+			return errors.Wrap(err, utils.AppError("failed to read response body"))
+		}
+		err = json.Unmarshal(body, &response)
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal response body")
+		}
+
+		// setup app file
+		appFile = text
+
+	} else {
+		var jsonValues []byte
+		url, path, err := utils.ParseURLGithub(file)
+		if err != nil {
+			return err
+		}
+
+		utils.Log.Debug(url)
+		utils.Log.Debug(path)
+
+		// save the app with Github URL
+		if path != "" {
+			jsonValues, _ = json.Marshal(map[string]interface{}{
+				"url":  url,
+				"path": path,
+				"save": true,
+			})
+		} else {
+			jsonValues, _ = json.Marshal(map[string]interface{}{
+				"url":  url,
+				"save": true,
+			})
+		}
+
+		req, err = utils.NewRequest("POST", appURL+"/"+sourceType, bytes.NewBuffer(jsonValues))
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		utils.Log.Debug("remote hosted app request success")
+		var response []*models.MesheryApplication
+		// failsafe (bad api call)
+		if resp.StatusCode != 200 {
+			return errors.Errorf("Response Status Code %d, possible Server Error", resp.StatusCode)
+		}
+		defer resp.Body.Close()
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return errors.Wrap(err, utils.AppError("failed to read response body"))
+		}
+		err = json.Unmarshal(body, &response)
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal response body")
+		}
+
+		// setup app file here
+		appFile = response[0].ApplicationFile
+	}
+
+	// Start conversion of app file to pattern file
+	jsonValues, _ := json.Marshal(map[string]interface{}{
+		"K8sManifest": appFile,
+	})
+
+	req, err = utils.NewRequest("POST", patternURL, bytes.NewBuffer(jsonValues))
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	var response []*models.MesheryPattern
+	// bad api call
+	if resp.StatusCode != 200 {
+		return errors.Errorf("Response Status Code %d, possible Server Error", resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, utils.AppError("failed to read response body"))
+	}
+
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return errors.Wrap(err, "failed to unmarshal response body")
+	}
+
+	utils.Log.Debug("application file converted to pattern file")
+
+	patternFile := response[0].PatternFile
+
+	req, err = utils.NewRequest("POST", appURL, bytes.NewBuffer([]byte(patternFile)))
+	if err != nil {
+		return err
+	}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer res.Body.Close()
+	body, err = io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	if res.StatusCode == 200 {
+		utils.Log.Info("app successfully imported")
+	}
+	utils.Log.Info(string(body))
+	return nil
+}
+
+func init() {
+	importCmd.Flags().StringVarP(&file, "file", "f", "", "Path/URL to app file")
```

Comment:
Are we still planning to have a name flag?

Reply:
No... we started to agree on using App IDs generated during import.